torch/csrc/inductor/aoti_torch/c/shim.h

```diff
@@ -487,6 +487,31 @@ AOTI_TORCH_EXPORT AOTITorchError aoti_torch_proxy_executor_call_function(
     int num_tensors,
     AtenTensorHandle* flatten_tensor_args);
 
+AOTI_TORCH_EXPORT void aoti_torch_check(
+    bool cond,
+    const char* func,
+    const char* file,
+    uint32_t line,
+    const char* msg);
+
+#ifdef STRIP_ERROR_MESSAGES
+#define AOTI_TORCH_CHECK(cond, ...)    \
+  aoti_torch_check(                    \
+      cond,                            \
+      __func__,                        \
+      __FILE__,                        \
+      static_cast<uint32_t>(__LINE__), \
+      TORCH_CHECK_MSG(cond, "", __VA_ARGS__));
```

Comment:
Wait, there is still `TORCH_CHECK_MSG`. Hmm, this is annoying, and seems like we have to pass var args. I wonder if it will be easier to just make `c10/util/Exception.h` header only (merging `c10/util/Exception.cpp` with `c10/util/Exception.h`). 

@ezyang , do you have any suggestion for this?

Reply:
By reading c10's utility functions, seems this TORCH_CHECK_MSG is header only. After it's expanded, it invokes ```torchCheckMsgImpl```:

https://github.com/pytorch/pytorch/blob/56364124af8fe148ba8b0c935571ebae6500f33b/c10/util/Exception.h#L451-L463

where ```::c10::str``` also seems to be header only:

https://github.com/pytorch/pytorch/blob/56364124af8fe148ba8b0c935571ebae6500f33b/c10/util/StringUtil.h#L114-L118

Reply:
I see. Then it should be ok for this PR. At some point, we need to come up with a small set of self-contained `c10/util` headers.

Reply:
> I see. Then it should be ok for this PR. At some point, we need to come up with a small set of self-contained `c10/util` headers.

Yes. That would be very helpful.